### DISCUSSION TITLE
Documenting how to use KamajiControlPlane provieder with CAPD

### DISCUSSION THREAD

**Discussion by kahirokunn** [clastix/cluster-api-control-plane-provider-kamaji#124] [Ideas]
## Summary
We need to create a CAPD (Cluster API Provider Docker) sample for the cluster-api-control-plane-provider-kamaji project.

## Background
Currently, it's challenging to perform local verifications. A CAPD sample would greatly facilitate this process.

## Objective
Creating a CAPD sample will provide the following benefits:
- Easier verification in local environments
- Faster testing for developers
- Better understanding of the project

## Tasks
1. Design a CAPD sample suitable for the cluster-api-control-plane-provider-kamaji project
2. Create the sample code
3. Develop documentation explaining how to use the sample

## Acceptance Criteria
- The CAPD sample works as expected
- Local verification can be performed easily
- Documentation is clear and usable by anyone

## Additional Notes
- Feel free to reference existing CAPD samples if available
- Consider the balance between sample complexity and ease of use

## Why is this important?
The primary reason for creating this CAPD sample is to make local verification easier. This will streamline the development process and allow for more efficient testing and debugging.

**Comment by prometherion** [comment]
I discussed this with @bsctl: since we would need to get some changes in the CAPD project, we picked KubeVirt as the development option, considering it could serve also production use cases as we saw with Cozystack /cc @kvaps 

Unless the community has the chance and the willingness to contribute, we're not going to provide support for CAPD.

**Reply by matofeder** [reply]
Could you please elaborate on the specific issues with CAPD? I'm wondering whether it's a fundamental limitation or if it's mainly about requiring changes upstream.

**Reply by prometherion** [reply]
It's a matter of upstream contribution, as it happened with several other providers.
